homeassistant/components/trafikverket_weatherstation/config_flow.py

```diff
@@ -0,0 +1,79 @@
+"""Adds config flow for Trafikverket Weather integration."""
+from __future__ import annotations
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY, CONF_MONITORED_CONDITIONS, CONF_NAME
+import homeassistant.helpers.config_validation as cv
+
+SENSOR_TYPES: dict[str, str] = {
+    "air_temp": "Air temperature",
+    "road_temp": "Road temperature",
+    "precipitation": "Precipitation type",
+    "wind_direction": "Wind direction",
+    "wind_direction_text": "Wind direction text",
+    "wind_speed": "Wind speed",
+    "wind_speed_max": "Wind speed max",
+    "humidity": "Humidity",
+    "precipitation_amount": "Precipitation amount",
+    "precipitation_amountname": "Precipitation name",
+}
+DOMAIN = "trafikverket_weatherstation"
+CONF_STATION = "station"
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): cv.string,
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Required(CONF_STATION): cv.string,
+        vol.Required(CONF_MONITORED_CONDITIONS, default=[]): cv.multi_select(
+            SENSOR_TYPES
+        ),
+    }
+)
+
+
+class TVWeatherConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Yale integration."""
+
+    VERSION = 1
+
+    entry: config_entries.ConfigEntry
+
+    async def async_step_import(self, config: dict):
+        """Import a configuration from config.yaml."""
+
+        self.context.update(
+            {"title_placeholders": {CONF_NAME: f"YAML import {DOMAIN}"}}
+        )
+        return await self.async_step_user(user_input=config)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            name = user_input[CONF_NAME]
+            api_key = user_input[CONF_API_KEY]
+            station = user_input[CONF_STATION]
+            conditions = user_input[CONF_MONITORED_CONDITIONS]
+
+            await self.async_set_unique_id(name)
+            self._abort_if_unique_id_configured()
```

Comment:
`unique_id` for config entries help to protect against accidentally setting up multiple config entries for an automatically discovered device, we don't need that here. If someone wants want 57 entries connecting to the same location, all with the same name, we don't judge.
```suggestion
```

Reply:
Okey. Me neither then :)

Reply:
Realized in testing that is I remove this it will import yaml config multiple time so I retain this in the code

Reply:
Move the check to import instead

Reply:
A `unique_id` is still not wanted.
The config flow has a helper `_async_abort_entries_match` which can be used to avoid multiple imports.

Example usage:
https://github.com/home-assistant/core/blob/2a2a20fcb3e19f1229aae8340286aa0e511bc0ff/homeassistant/components/broadlink/config_flow.py#L298-L301

Reply:
Didn't know this function existed so thanks for that. I have now changed according to the example and should now be fine then. Thanks for reviewing it and hope it can get merged shortly.